satellite/metainfo/validation.go

```diff
@@ -332,6 +334,12 @@ func (endpoint *Endpoint) validatePointer(ctx context.Context, pointer *pb.Point
 			if piece.NodeId != limit.StorageNodeId {
 				return Error.New("piece NodeID != order limit NodeID")
 			}
+
+			pieceNums[piece.PieceNum] = struct{}{}
+			nodeIds[piece.NodeId] = struct{}{}
+		}
+		if len(pieceNums) != len(remote.RemotePieces) || len(nodeIds) != len(remote.RemotePieces) {
+			return Error.New("invalid number of unique pieces")
```

Comment:
If this error happens, it might be hard to debug the root cause.

Can we move the check within the for-cycle and return an error as soon as a duplicated piece number or node ID is detected? The returned error should include the duplicated piece number or node ID.

Reply:
done